contracts/test/benefactorLeave.js

```diff
@@ -0,0 +1,200 @@
+      from: accounts[2],
+    });
+
+    let depositedDaiUser = await noLossDao.depositedDai.call(accounts[2]);
+    let totalDai = await noLossDao.totalDepositedDai.call();
+
+    assert.equal(applicationAmount, depositedDaiUser.toString());
+    assert.equal(applicationAmount, totalDai);
+
+    await expectRevert(
+      noLossDao.withdrawProposal({
+        from: accounts[2],
+      }),
+      'Benefactor only eligible to receive funds in later iteration'
+    );
+
+    await time.increase(time.duration.seconds(1801)); // increment to iteration 1
+    await noLossDao.distributeFunds();
+
+    await expectRevert(
+      noLossDao.withdrawProposal({
+        from: accounts[2],
+      }),
+      'Benefactor only eligible to receive funds in later iteration'
+    );
+
+    await time.increase(time.duration.seconds(1801)); // increment to iteration 2
+    await noLossDao.distributeFunds();
+
+    await expectRevert(
+      noLossDao.withdrawProposal({
+        from: accounts[2],
+      }),
+      'Benefactor only eligible to receive funds in later iteration'
+    );
+
+    await time.increase(time.duration.seconds(1801)); // increment to iteration 3
+    await noLossDao.distributeFunds();
+
+    noLossDao.withdrawProposal({
+      from: accounts[2],
+    });
+
+    // Once withdrawn later
+    let depositedDaiUser2 = await noLossDao.depositedDai.call(accounts[2]);
+    assert.equal('0', depositedDaiUser2.toString());
+  });
+
+  it('NoLossDao:benefactorLeave. Benefactor create withdraw, create withdraw... ', async () => {
+    let mintAmount = '60000000000';
+
+    await dai.mint(accounts[2], mintAmount);
+    await dai.approve(noLossDao.address, mintAmount, {
+      from: accounts[2],
+    });
+    await noLossDao.createProposal('Some IPFS hash string', {
+      from: accounts[2],
+    });
+
+    await time.increase(time.duration.seconds(1801)); // increment to iteration 1
+    await noLossDao.distributeFunds();
+
+    await time.increase(time.duration.seconds(1801)); // increment to iteration 2
+    await noLossDao.distributeFunds();
+
+    await time.increase(time.duration.seconds(1801)); // increment to iteration 3
+    await noLossDao.distributeFunds();
+
+    noLossDao.withdrawProposal({
+      from: accounts[2],
+    });
+
+    await noLossDao.createProposal('Some IPFS hash string again', {
+      from: accounts[2],
+    });
+
+    await time.increase(time.duration.seconds(1801)); // increment to iteration 1
+    await noLossDao.distributeFunds();
+
+    await time.increase(time.duration.seconds(1801)); // increment to iteration 2
+    await noLossDao.distributeFunds();
+
+    await expectRevert(
+      noLossDao.withdrawProposal({
+        from: accounts[2],
+      }),
+      'Benefactor only eligible to receive funds in later iteration'
+    );
+
+    await time.increase(time.duration.seconds(1801)); // increment to iteration 3
+    await noLossDao.distributeFunds();
+
+    noLossDao.withdrawProposal({
```

Comment:
You need to make this an await, and you need to check that it worked (it could also be reverting and you wouldn't know).